fix: add eslintIgnore in package.json

ThangHuuVu opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free
video course helpful: https://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

## Reasoning 💡

The `lint` command is returning (38) errors: 
```
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: app/components/access-denied.js.
The file must be included in at least one of the projects provided
...
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/client/__tests__/client-provider.test.js.
The file must be included in at least one of the projects provided
...
```
I fix this by adding the `eslintIgnore` object in package.json with the value get from `tsconfig.json` `exclude`. 

<!-- What changes are being made? What feature/bug is being fixed here? -->

## Checklist 🧢

<!-- Feel free cross items ( like this `~[] item~` ) if they're irrelevant to your changes.

To check an item, place an `x` in the box like so: `- [x] Documentation`. -->

- ~[ ] Documentation~
- ~[ ] Tests~
- [x] Ready to be merged

<!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->

## Affected issues 🎟

<!--
Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

If you write `"Fixes"` or `"Closes"` before the issue link like so:

```
Fixes #359
```

the connected issue will be automatically closed once the PR is merged and hence help with maintenance of the library 😊

-->

ThangHuuVu committed:
fix: add eslintIgnore in package.json

balazsorban44 commented on package.json:
```diff
@@ -174,6 +174,14 @@
     ]
   },
+  "eslintIgnore": [
+    "./*.js",
+    "./*.d.ts",
+    "app",
+    "**/tests",
+    "**/__tests__",
+    "config"
```
```suggestion
    "./*.d.ts",
    "**/tests",
    "**/__tests__",
```

Ideally, I would like ESLint to report in most of these directories.

ThangHuuVu replied:
Since we want to run the lint command on `app`, `src` and `config`, but at the same time want `tsc` to compile files in `src` only, I introduced a separate `tsconfig.eslint.json` file for `eslint`, extending from the base `tsconfig.json` file. Do you think this makes good sense?

balazsorban44 replied:
I'll have to check. I generally don't run the lint command, just let VSC warn me and I fix the warnings immediately.

ThangHuuVu replied:
hmm, how do you think about adding lint-staged and husky to lint before committing?

balazsorban44 replied:
Considering the low amount of external PRs, I never really thought this would be necessary. We tried adding `husky` once, but it had issues and got reverted. (#2156, https://github.com/nextauthjs/next-auth/pull/2130#issuecomment-858145520)

Since it's mainly me who creates code, it seemed unnecessary until this point. If we can do it in a reliable way, I guess it would be alright.

ThangHuuVu replied:
Got it. After checking, please let me know if there's anything that should be changed in this PR.

ThangHuuVu committed:
Let eslint runs in app, config + js files

ThangHuuVu committed:
Add a separate tsconfig.eslint.json file

We want to run the lint command on `app`, `src` and `config`, but at the same time want `tsc` to compile files in `src` only. A separate `tsconfig.eslint.json` is a suitable solution to satisfy both `eslint` and `tsc`: https://github.com/typescript-eslint/typescript-eslint/blob/04d1f3e5492a26e6f81002e0b89d58ba7cf60adb/packages/parser/README.md

ThangHuuVu committed:
Merge branch 'main' into main

balazsorban44 commented on app/pages/api/auth/[...nextauth].ts:
```diff
@@ -1,5 +1,5 @@
 import NextAuth, { NextAuthOptions } from "next-auth"
-import EmailProvider from "next-auth/providers/email"
+// import EmailProvider from "next-auth/providers/email"
```
Could this be commented out by default? I often use the dev app to check something, without an adapter
